### DISCUSSION TITLE
Discrepancy in Variant Pair and Misorientation Angle Data

### DISCUSSION THREAD

**Discussion by ghost** [ORTools4MTEX/ORTools#74] [Q&A] [answered]
Hi everyone,
I've been working on plotting variant pairs with V1 and the misorientation angles between the BCC phases in steel. I've attached my plots here.
However, I have encountered a puzzling discrepancy: in the variant pair plot, the V1-V4 pair, which corresponds to a low-angle boundary, shows a high fraction. 
Conversely, the fraction of low-angle boundaries in the misorientation angle plot is notably small.
Besides that i've plotted the orientation map for both V1 and V4 (together which is attached here) and again i can't see a high fraction of V1-V4 pairs. However, I see in the variant-pair distribution plot, V1-V4 has the highest fraction.
I would appreciate your insights on why there might be such a difference in the representation of low-angle boundaries between these two analyses. 
Also, I've attached the commands that I used for my EBSD data. I couldn't attach my data because it was larger than 25 MB.
Thank you for your time and assistance.
![image](https://github.com/user-attachments/assets/ac84d63a-917f-400f-8970-af3cfba7025d)
![image](https://github.com/user-attachments/assets/85b5bdc3-ff99-464e-b14f-550f49387fee)
![image](https://github.com/user-attachments/assets/17371487-08ef-492c-ae8f-acff9927ac4d)
![Variant 1 4](https://github.com/user-attachments/assets/2d9bf54b-df6b-41bd-a657-3492b4d1c0a2)

```
ebsd = EBSD.load('950  - Iso 400.ctf')
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
ebsd(grains(grains.grainSize < 3)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree);
grains = smooth(grains,5)

%%% This line is responsible for plotting misorientation angle
gB = grains.boundary('AlphaP','AlphaP')
plotAngleDistribution(gB.misorientation)
%%%

job = setParentGrainReconstructor(ebsd,grains,Ini.cifPath);
KS = orientation.KurdjumovSachs(job.csParent,job.csChild);
job.p2c = KS;
job.calcParent2Child;
job.calcVariantGraph('threshold',4*degree,'tolerance',3.5*degree);
job.clusterVariantGraph('includeSimilar');
job.calcParentFromVote('minProb',0.5)
job.calcGBVotes('p2c','reconsiderAll')
job.calcParentFromVote
job.mergeSimilar('threshold',7.5*degree);
job.mergeInclusions('maxSize',150);
job.calcVariants;

%%%This line is responsible for plotting variant pairs with equivalant Id
variantBoundaries_map = plotMap_variantPairs(job,'linewidth',1.5);
plotMap_variantPairs(job,'linewidth',1.5)
%%%

%%%This line is responsible for plotting variant pairs with V1
[newGrains,~] = computeVariantGrains(job);
newGrains = newGrains(job.csChild);
vGroupIds = {[1 2],...
    [1 3; 1 5],...
    [1 4],...
    [1 6],...
    [1 7],...
    [1 8],...
    [1 9; 1 19],...
    [1 10; 1 14],...
    [1 11; 1 13],...
    [1 12; 1 20],...
    [1 15; 1 23],...
    [1 16],...
    [1 17],...
    [1 18; 1 22],...
    [1 21],...
    [1 24]};
	
out1 = computeGrainPairs(newGrains,'variant','group',vGroupIds,'plot');
```

**Comment by frankNiessen** [comment]
Hi, 

I think we would need access to your dataset in order to investigate this further. 

Cheers
Frank

**Comment by AzdiarGazder** [comment]
Hi @bamobe,

As requested by @frankNiessen above and me in an earlier email to you, please upload the full script and your EBSD data set to a website that hosts files and paste the weblink to those files in the discussion here. Then we can download the files and help diagnose your issue.

Warm regards,
Azdi

**Comment by pj-decastro** [comment]
Hello all,

I had a similar event with my dataset. 
I noticed that, for my steel (not Martensitic), the newgrains function also attributes new boundaries with very low angle (<4°) as V1-V4. To have a comparison, I plotted the Takayama pairing approach for the transformedGrains, and the V1-V4 frequency was much lower.

New grains:
![Pairing_New-grains](https://github.com/user-attachments/assets/0e5bec0b-692a-486f-905d-0d3eab40dfde)
Transformed grains:
![Pairing_Transformed_grains](https://github.com/user-attachments/assets/1ff8f411-7e31-46d0-99b9-9423c3caeecb)

Perhaps the newgrains function is attributing small misorientation fluctuations to V1-V4 pairings?

Best,
Pedro

**Reply by frankNiessen** [reply, marked as answer]
I think you are absolutely right here Pedro, thanks for pointing this out. 

So while for the K-S orientation relationship the V1-V4 boundary has a misorientation angle of 10.53 degree, this reduces quite significantly when fitting the OR to the microstructure (depending on the microstructure of course). In the standard MTEX `martensite` example, the misorientation angle reduces to 4.2 degree. 

Martensitic laths and are not clean boundaries with a nice and constant misorientation angle; they twist and turn quite a bit by several degrees along their long axis, see this paper:

> Sandvik, B.P.J., Wayman, C.M. Characteristics of lath martensite: Part I. crystallographic and substructural features. Metall Trans A 14, 809–822 (1983). https://doi.org/10.1007/BF02644284

When reconstructing the grains using the `calcGrains` command in MTEX, we identify regions of EBSD pixels that are below a misorientation angle treshold of say 3 degree. A lot of the substructure of martensite will thus not be picked up using this approach. 

After having reconstructed the parent microstructure, we can determine the variant ID for each EBSD pixel, we know the OR and the reconstructed parent grain orientation after all. We can then reconstruct the grains again, but this time not with a threshold value as in `calcGrains`, but by identifying areas of common variant id. This is the method that is used in `computeVariantGrains`. 

Evidently, the latter leads to a refinement of the microstructure to quite a lot more grains (for the MTEX `martensite` dataset we go from ~7,000 to ~31,000 grains). The boundaries that are being created here are almost entirely V1-V4 boundaries, which should explain the high weighting in the histogram. I do think that this type of grain segmentation makes sense and gives a more refined description of the microstructure.

But in case the user does not want this, all analysis can be executed on the grains computed with `calcGrains` as well, which should give a lower V1-V4 weight.

**Reply by pj-decastro** [reply]
Hello Frank, thank you for the detailed answer.
What I was worried about is if the application of the NewGrains (computeVariantGrains) with a refined OR is attributing very low misorientations, < 2°, to new variant pairs... I believe that these very low misorientations might be just a "background" fluctuation within what really is just one variant.
What are your thoughts on this? 
Perhaps we could have a lower threshold to avoid such classifications embedded in the function?

Edit: here are the misorientation distribution of the correspondent EBSD maps of my post above:
![grafik](https://github.com/user-attachments/assets/1ca0b981-7dd6-4217-a3e1-6749d951b040)
I.e. the increase in V1-V4 variants is due to very low misorientations, below 1° in the picture.

**Reply by frankNiessen** [reply]
Hi again, 

You are right that the application of the newGrains does lead to a lot more low-angle misorientations, and this is well intended. 

The question is whether these are induced by background noise are are real. This is really for you to decide, the user, and if you feel like this is background noise you should not use it. 

I am confident that in a properly acquired map of lath martensite, you will be able to identify V1-V4 pairs that would fall underneath the radar using conventional grain reconstruction. 

If your dataset is not very clean or if you are not looking for low-angle boundaries, I would not recommend using this method.
